[ZHANGTIANYAO1] iP

ZHANGTIANYAO1 opened the pull request:
### Saopig ChatBot
---

> Hi there! I'm a gentle and proud cartoon girl ChatBot, here to spread joy and help you with your tasks in the most delightful way! 🌸✨  – Simulated personality by [ChatGPT](https://chat.openai.com/)

Saopig ChatBot is here to assist you with:
- providing cheerful encouragement along the way
- managing tasks
- make sure everything is as lovely and efficient as a well-tended garden! 📚🌼✨

All you need to do is,
1. download it from [here](https://github.com/ZHANGTIANYAO1/ip/releases/download/A-Jar/saopig.jar)
2. open the cmd or terminal
3. cd to your download directory
4. run `java -jar saopig.jar`
5. add your tasks.
6. enjoy managing your tasks!

And it is **FREE & OPENSOURCE!** 

Features:
- [x] Managing todo tasks
- [x] Managing deadline tasks
- [x] Managing event tasks
- [x] Mark&Unmark tasks
- [x] Use the time filter tasks
- [x] GUI
---
If you are a Java programmer, you can use it to practice Java too. Here's the Saopig class:

```java
public class Saopig {
    private TaskList taskList = new TaskList();
    private Ui ui = new Ui();
    private Storage storage = new Storage(this.ui, this.taskList);

    /**
     * Runs the Saopig chatbot.
     */
    public static void main(String[] args) {
        new Saopig().run();
    }

    /**
     * Runs the Saopig chatbot until the user exits.
     */
    private void run() {
        ui.showWelcome();
        boolean isExit = false;
        while (!isExit) {
            try {
                String fullCommand = ui.readCommand();
                Command c = Parser.parse(fullCommand);
                c.execute(this.taskList, this.ui, this.storage);
                isExit = c.isExit();
                ui.printLine();
            } catch (SaopigInvaildSizeException e) {
                ui.showError("Unknown command error.");
                ui.printLine();
            }
        }
        ui.showGoodbye();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ZHANGTIANYAO1 committed:
Level-0

ZHANGTIANYAO1 committed:
implement Level 1. Echo and add enums for command

ZHANGTIANYAO1 committed:
Level 2

ZHANGTIANYAO1 committed:
leve 3 and level 4

ZHANGTIANYAO1 committed:
fix Font error, config Automated Text UI Testing

ZHANGTIANYAO1 committed:
add custom exception

ZHANGTIANYAO1 committed:
Handle Errors，update A-TextUiTesting

ZHANGTIANYAO1 committed:
Handle Errors，update A-TextUiTesting

ZHANGTIANYAO1 committed:
Level 6. Delete, add more error handles

ZHANGTIANYAO1 committed:
Formatted code
Level-7 Save

ZHANGTIANYAO1 committed:
Add more test cases

ZHANGTIANYAO1 committed:
Merge branch 'branch-Level-7'

ZHANGTIANYAO1 committed:
Level 8 Dates and Times
Update input file
Add ListTaskOnDate Command

ZHANGTIANYAO1 committed:
Merge branch 'branch-Level-8'

ZHANGTIANYAO1 committed:
A-MoreOOP

ZHANGTIANYAO1 committed:
Merge branch 'A-MoreOOP'

ZHANGTIANYAO1 committed:
A-Packages

ZHANGTIANYAO1 committed:
Merge branch 'A-Packages'

ZHANGTIANYAO1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

# Conflicts:
#	docs/README.md

ZHANGTIANYAO1 committed:
Use Gradle
Add JUnit Tests

ZHANGTIANYAO1 committed:
Create a JAR File

ZHANGTIANYAO1 committed:
Merge branch 'add-gradle-support'

ZHANGTIANYAO1 committed:
Add JavaDoc comments

ZHANGTIANYAO1 committed:
Give users a way to find a task by searching for a keyword

ZHANGTIANYAO1 committed:
Tweak the code to comply with a coding standard

ZHANGTIANYAO1 committed:
Merge branch 'branch-Level-9'

ZHANGTIANYAO1 committed:
Merge branch 'branch-A-JavaDoc'

ZHANGTIANYAO1 committed:
Merge branch 'branch-A-CodingStandard'

ZHANGTIANYAO1 committed:
fix error

ZHANGTIANYAO1 committed:
add checkstyle
fix style

ZHANGTIANYAO1 committed:
Add GUI
Fix bugs

ZHANGTIANYAO1 committed:
Merge branch 'branch-A-GUI'

ibnu2651 commented on src/main/java/saopig/command/Command.java:
```diff
@@ -0,0 +1,27 @@
+     * @param tasks   TaskList object.
+     * @param ui      Ui object.
+     * @param storage Storage object.
+     * @return
```
I think you may have missed this out?

ibnu2651 commented on src/main/java/saopig/command/ListCommand.java:
```diff
@@ -0,0 +1,162 @@
+     * @param input    The input.
+     * @param taskList The task list.
+     * @param ui       The user interface.
+     * @return response to the user.
```
I think the first word after return should start with a capital letter? Maybe you can clarify and make some changes since it's a bit inconsistent in the whole codebase.

ibnu2651 commented on src/test/java/saopig/command/ListCommandTest.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @Test
+    void testListTasksOnDateInvalidInputLength() {
```
Maybe for some of the tests with longer and cryptic names you could use the alternative naming format unitBeingTested_descriptionOfTestInputs_expectedOutcome? So the reader has a clearer idea of the test and expected outcome.

ibnu2651 reviewed (commented):
I think this is a very thorough and well executed piece of work and I genuinely could not find much to improve on. Keep up the good work!

ZHANGTIANYAO1 commented:
> I think this is a very thorough and well executed piece of work and I genuinely could not find much to improve on. Keep up the good work!

Thanks for your review I will fix those nits you point out

ZHANGTIANYAO1 committed:
fix nits

ronnnnnnnnn reviewed (commented):
LGTM. Honestly cannot find any issues as far as code quality is concerned. Good work.

ZHANGTIANYAO1 committed:
Use Assertions in the Command Package

Modified the gradle setting to enable Java' assertions at runtime

Assertions have been added to most of the Command classes to help
detect whether the type index conforms to the set optional range.

Add assertions after processing the input length to determine if
the input still does not conform to the expected specification.

ZHANGTIANYAO1 committed:
Merge pull request #1 from ZHANGTIANYAO1/A-Assertions

Use Assertions in the Command Package

ZHANGTIANYAO1 committed:
A-CodeQuality

ZHANGTIANYAO1 committed:
Merge pull request #2 from ZHANGTIANYAO1/heads/A-CodeQuality

A-CodeQuality

ZHANGTIANYAO1 committed:
A-CI

ZHANGTIANYAO1 committed:
fix style

ZHANGTIANYAO1 committed:
Update test

ZHANGTIANYAO1 committed:
Fix test code style

ZHANGTIANYAO1 committed:
Update gradle.yml

ZHANGTIANYAO1 committed:
Update gradle.yml

ZHANGTIANYAO1 committed:
update test code

ZHANGTIANYAO1 committed:
Merge branch 'master' of https://github.com/ZHANGTIANYAO1/ip

ZHANGTIANYAO1 committed:
Support a way to easily edit details of items

ZHANGTIANYAO1 committed:
Merge tag 'C-Update'

ZHANGTIANYAO1 committed:
update readme intro

ZHANGTIANYAO1 committed:
Better GUI

ZHANGTIANYAO1 committed:
Merge branch 'A-BetterGui'

ZHANGTIANYAO1 committed:
A-UserGuide
imporve code style

ZHANGTIANYAO1 committed:
Merge branch 'A-UserGuide'

ZHANGTIANYAO1 committed:
fix image nits

ZHANGTIANYAO1 committed:
upload image

ZHANGTIANYAO1 committed:
fix nits for release

ZHANGTIANYAO1 committed:
Update README.md

ZHANGTIANYAO1 committed:
Update README.md

ZHANGTIANYAO1 committed:
Fix listOnDate command

Fix list task display

Fix bye command

ZHANGTIANYAO1 committed:
Merge branch 'master' of https://github.com/ZHANGTIANYAO1/ip

ZHANGTIANYAO1 committed:
bug fix: Fix ListCommandTest.java

ZHANGTIANYAO1 committed:
Update the UG for update command

Fix too long methods

More abstractions

ZHANGTIANYAO1 committed:
Update header comments
